module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -238,11 +238,23 @@ public function convertToQueryStringParameterArray()
         }
 
         if (isset($this->facetFilters) && 0 < count($this->facetFilters)) {
-            $formatFilter = function ($raw) {
-                [$field, $value] = explode(':', $raw, 2);
-                return $field . ':' . static::escapeSpecialCharacters($value);
-            };
-            $qs['facetfilter'] = array_map($formatFilter, $this->facetFilters);
+            $filterId = 1;
+            $qs['facetfilter'] = [];
+            foreach ($this->facetFilters as $field => $values) {
+                $field = str_replace('~', '', $field);
+                $values = array_map(fn ($value) => static::escapeSpecialCharacters($value), $values);
+                $operator = $this->getFacetOperator($field);
+                if ('OR' == $operator) {
+                    $valuesString = implode(',', array_map(fn ($value) => "{$field}:{$value}", $values));
+                    $qs['facetfilter'][] = "{$filterId},{$valuesString}";
+                    $filterId++;
+                } else {
+                    foreach ($values as $value) {
+                        $qs['facetfilter'][] = "{$filterId},{$field}:{$value}";
+                        $filterId++;
+                    }
+                }
+            }
```

Comment:
I question the wisdom of supporting GET requests at all with the EDS API; as the EDS.ini says, we're likely to run into query length problems eventually, and the query string becomes increasingly unreadable with all of the punctuation for filters that has to be URL encoded. But since we do support GET, I'm doing that here.

Reply:
If memory serves, GET support has been useful in the (fairly distant) past for troubleshooting purposes, since it is significantly different from POST and can be a double-check when things aren't behaving as expected. That's not necessarily a compelling reason to keep it, and I wouldn't necessarily object to deprecating the option if you think it's best... but at any rate, that's the only justification I can think of.

Reply:
> GET support has been useful in the (fairly distant) past for troubleshooting purposes, since it is significantly different from POST and can be a double-check when things aren't behaving as expected.

Actually that is somewhat compelling.  Anyway I don't suggest removing GET support in this PR, just putting the thought down for the future.